corehq/apps/analytics/templates/analytics/initial/gtm.html

```diff
@@ -0,0 +1,13 @@
+{% load hq_shared_tags %}
+{% initial_analytics_data 'gtm.apiId' ANALYTICS_IDS.GTM_ID %}
+{% if request.couch_user %}
+  {% initial_analytics_data 'gtm.userId' request.couch_user.userID %}
+  {% initial_analytics_data 'gtm.userIsDimagi' request.couch_user.is_dimagi %}
+  {% initial_analytics_data 'gtm.userIsCommCare' request.couch_user.is_commcare_user %}
+{% endif %}
+{% if domain %}
+  {% initial_analytics_data 'gtm.domain' domain|escapejs %}
```

Comment:
why we are using escapejs here?

Reply:
Thanks for pointing this out. 
I reused this code from [google.html](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/analytics/templates/analytics/initial/google.html#L11). I believe it was added to avoid XSS attack . After looking more into it and don't think it is needed here as we are capturing domain from the url and it should not allow js mixed into it. 
Removed it in https://github.com/dimagi/commcare-hq/commit/ac89b8aba537dd094044eaf1b2880d2c9b7318f2

Reply:
Oh, in that case I say we should retain it.

I assume its possible for someone to put anything in the domain part in the url & HQ will look for it. No actual domain page would load but I assume this part still gets added to the page, hence the escapejs.

Reply:
Just a spitball suggestion (feel free to ignore): 
I agree with Manish's concern about someone manually changing the domain part of the url, so either
1. keep the escapejs filter
2. access the domain through a way we know it's right, e.g. `request.couch_user.domain` (maybe even `request.domain`?)

Option 2 seems least pretty though.

Reply:
If the domain is invalid, the page should have 404ed long before this point.

Reply:
> If the domain is invalid, the page should have 404ed long before this point.

Yes, this was tested and it is not possible to add script from the url. 

@Charl1996 Correct me if I am wrong, my understanding is that domain for `request` or `request.couch_user` is still obtained from url ?  (A user may be associated with multiple domains.)